### PR TITLE
Add Server Links

### PR DESCRIPTION
**Pull request by Commandcracker**
## Checklist
Things need to be done before this Pull Request can be merged.

- [x] Code is well-formatted and adheres to project style guidelines: `cargo fmt`
- [x] Code does not produce any clippy warnings `cargo clippy`

### CODE REVIEW

File: pumpkin/src/world/mod.rs [hunk 1]
```diff
@@ -203,6 +204,68 @@ impl World {
...
+                links.push(Link::new(&Label::BuiltIn(LinkType::News), news))
+            }
+
+            let announcements = &ADVANCED_CONFIG.server_links.announcements;
+            if !announcements.is_empty() {
+                links.push(Link::new(&Label::BuiltIn(LinkType::Announcements), announcements))
+            }
+
+            for (key, value) in &ADVANCED_CONFIG.server_links.custom {
+                // add
```
Comment (DataM0del) [inline pumpkin/src/world/mod.rs:256]:
```suggestion
                todo!(format!("Setting custom links hasn't been implemented yet! K: {key} V: {value}"))
```
Makes clippy not scream at you for not using `key` and `value`.

**Comment by DataM0del** [comment]
See Commandcracker#1

File: pumpkin/src/world/mod.rs (hunk 1 above)
Comment (DataM0del) [reply pumpkin/src/world/mod.rs:256]: I implemented it in this PR: Commandcracker#1

**Review by DataM0del — APPROVED** [review]
I see why you didn't make them `Option`s now.

File: pumpkin/src/world/mod.rs [hunk 2]
```diff
@@ -203,6 +204,68 @@ impl World {
         player.send_permission_lvl_update().await;
         client_cmd_suggestions::send_c_commands_packet(&player, command_dispatcher).await;
 
+        if ADVANCED_CONFIG.server_links.enabled {
```
Comment (Commandcracker) [inline pumpkin/src/world/mod.rs:207]: Where should this logic be placed ? I don't think this is the right place.

File: pumpkin/src/world/mod.rs (hunk 1 above)
Comment (Commandcracker) [inline pumpkin/src/world/mod.rs:256]: Got some lifetime and borrowing issue here, would be nice if a rust pro could explain it to me 👉👈

Comment (Commandcracker) [reply pumpkin/src/world/mod.rs:297]:
```rs
for (key, value) in &ADVANCED_CONFIG.server_links.custom {
    links.push(Link::new(&Label::TextComponent(TextComponent::text(key)), value))
}
```

```
error[E0716]: temporary value dropped while borrowed                                                                                                                                                        
   --> pumpkin\src\world\mod.rs:297:39
    |
297 |                 links.push(Link::new(&Label::TextComponent(TextComponent::text(key)), value))
    |                 -----                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ creates a temporary value which is freed while still in use
    |                 |
    |                 borrow later used here
298 |             }
    |             - temporary value is freed at the end of this statement
    |
    = note: consider using a `let` binding to create a longer lived value

For more information about this error, try `rustc --explain E0716`.
error: could not compile `pumpkin` (bin "pumpkin") due to 1 previous error
```

File: pumpkin/src/world/mod.rs [hunk 3]
```diff
@@ -14,6 +14,7 @@ use crate::{
     error::PumpkinError,
     server::Server,
 };
+use itertools::Itertools;
```
Comment (DataM0del) [inline pumpkin/src/world/mod.rs:17]:
```suggestion
```
What is this supposed to be here for?

Comment (Commandcracker) [reply pumpkin/src/world/mod.rs:17]: Because I tried around some stuff

**Comment by Snowiiii** [comment]
There are 2 Server link Packets, One in the Config State and one in the Play State. You can keep the Play one, it will be useful later. But i think it would generally be better if we send this at the Config state

**Comment by Snowiiii** [comment]
LGTM.
Thanks @Commandcracker

**Pull request merged**